### PR TITLE
Use precompiled handlebars

### PR DESCRIPTION
**Pull request by sharidas**
This change will help to use
precompiled handlebars.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

### CODE REVIEW

**Comment by sharidas** [comment]
This PR is same as https://github.com/owncloud/impersonate/pull/42. The difference is that this change is against master and not on top of the logout change made in the commit mentioned (https://github.com/owncloud/impersonate/pull/42 ). I have versified this change against 10.0.2 version. And it worked.

File: Makefile [hunk 1]
```diff
@@ -19,11 +29,27 @@ endif
...
 #
 .PHONY: dist
-dist: build-src
+dist: build-dep js-templates build-src
+
+%.js: $(template_src)
+		$(HANDLEBARS) -n "$(js_namespace)" $* -f $@
+
+build-dep:
+		make build-dep
```
Comment (PVince81) [inline Makefile]: It doesn't make sense to have the same build target twice here, please remove.

Also, from within a Makefile you usually don't need to run `make` at all. You need to use the dependencies feature of Makefiles to make it run other targets before the current one.

File: Makefile [hunk 2]
```diff
@@ -43,6 +69,7 @@ ifdef CAN_SIGN
 		build/$(appname)
 	cp settings-admin.php build/$(appname)
 	$(sign) --path="$(CURDIR)/build/$(appname)"
+	rm -f build/$(appname)/js/templates/*.handlebars
```
Comment (PVince81) [inline Makefile]: please remove **before** signing

**Comment by sharidas** [comment]
Removed the change requested for build-dep. Also removed the handlebar files before signing the app.

**Comment by sharidas** [comment]
Closing this PR. Duplicate of https://github.com/owncloud/impersonate/pull/42

**Pull request closed without merging**